src/RandMsg.chpl

```diff
@@ -296,9 +297,82 @@ module RandMsg
         randLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),repMsg);
         return new MsgTuple(repMsg, MsgType.NORMAL);
     }
-    
+
+    proc permutationMsg(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab): MsgTuple throws {
+        const pn = Reflection.getRoutineName();
+        var rname = st.nextName();
+        const name = msgArgs.getValueOf("name");
+        const xName = msgArgs.getValueOf("x");
+        const size = msgArgs.get("size").getIntValue();
+        const dtypeStr = msgArgs.getValueOf("dtype");
+        const dtype = str2dtype(dtypeStr);
+        const state = msgArgs.get("state").getIntValue();
+        const isDomPerm = msgArgs.get("isDomPerm").getBoolValue();
+
+        randLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
+                                "name: %? size %i dtype: %? state %i isDomPerm %?".doFormat(name, size, dtypeStr, state, isDomPerm));
+
+        st.checkTable(name);
+
+        proc permuteHelper(type t, st: borrowed SymTab) throws {
```

Comment:
Why does `st` need to be passed in as an argument?

Reply:
it doesn't, it's just force of habit cause we are getting and putting things in symbol table. But ur completely right-- since this is a nested proc, we have access to `st` from the outer scope and don't have to pass it in